mule-extensions-api/src/main/java/org/mule/runtime/extension/api/client/source/SourceResultCallback.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.client.source;
+
+import org.mule.api.annotation.Experimental;
+import org.mule.runtime.api.meta.model.source.SourceCallbackModel;
+import org.mule.runtime.api.meta.model.source.SourceModel;
+import org.mule.runtime.extension.api.client.ExtensionsClient;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.sdk.api.annotation.MinMuleVersion;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+
+/**
+ * Callback used by {@link ExtensionsClient} to notify of new {@link Result results} produced by a source created through the
+ * {@link ExtensionsClient#createSource(String, String, Consumer, Consumer)} method.
+ * <p>
+ * Clients will obtain the produced value by calling the {@link #getResult()} method.
+ * <p>
+ * After processing the result, clients <b>MUST</b> invoke either {@link #completeWithSuccess(Consumer)} or
+ * {@link #completeWithError(Throwable, Consumer)}, so that the source can finish the processing of each individual result. Refer
+ * to those method's javadocs for further details on their semantics.
+ * <p>
+ * <b>NOTE:</b> Experimental feature. Backwards compatibility not guaranteed.
+ *
+ * @param <T> the generic type of the result output value
+ * @param <A> the generic type of the result attributes value
+ * @since 1.6.0
+ */
+@Experimental
+@MinMuleVersion("4.6.0")
+public interface SourceResultCallback<T, A> {
```

Comment:
What about, SourceResultProcessor? or Handler.

I think that this holds the callback methods but is not really a callback.